x-pack/filebeat/input/http_endpoint/crc.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package http_endpoint
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Validator func(crcValidator, mapstr.M) (int, string, error)
+
+type crcValidator struct {
+	provider  string    // Name of the webhook provider
+	key       string    // Key to identify CRC requests (optional)
+	value     string    // Value of the field to identify CRC requests (optional)
+	challenge string    // Key of the challenge token
+	secret    string    // Webhook's secret token
+	validator Validator // Function to process the CRC request
```

Comment:
This is an action, so I'd rename it `validate`, then the call-site reads as a verb. Alternatively, add a `validate` method to `crcValidator` that checks whether `validator` is nil and does not call if it is. Further alternative is to keep the `crcValidator` as a pointer, then the `validate` method can be
```
func (v *crcValidator) validate(m mapstr.M) (status int, resp string, _ error) {
    return v.validator(v, m)
}
```
with clarity benefits at the call site.

(requires change to `Validator`'s signature).